src/extension.ts

```diff
@@ -47,6 +48,67 @@ export function activate(context: ExtensionContext) {
     })
   );
 
+  // When extension is started, loop for each text documents which are opened to update their language ID.
+  workspace.textDocuments.forEach(document => {
+    updateLanguageId(document);
+  });
+  // When a text document is opened,  update their language ID.
+  context.subscriptions.push(
+    workspace.onDidOpenTextDocument((document) => {
+      updateLanguageId(document);
+    })
+  );
+
+  /**
+   * Update if required the language ID to 'quarkus-properties' if needed.
+   *
+   * @param document the text document.
+   */
+  async function updateLanguageId(document: TextDocument) {
+    if (document.fileName.endsWith('application.properties')) {
+      if (document.languageId === 'quarkus-properties') {
+        // the language ID is already quarkus-properties do nothing.
+        return;
+      }
+      tryToForceLanguageId(document, 'quarkus-properties');
+    } else if (document.fileName.endsWith('application.yaml') || document.fileName.endsWith('application.yml')) {
+      if (document.languageId === 'yaml') {
+        // the language ID is already yaml do nothing.
+        return;
+      }
+      tryToForceLanguageId(document, 'yaml');
+    }
+  }
+
+  /**
+   * Try to force the document language ID to the given language ID.
+   *
+   * @param document the text document.
+   * @param languageId the language ID.
+   */
+  async function tryToForceLanguageId(document: TextDocument, languageId: string) {
+    const propertiesLanguageMismatch: PropertiesLanguageMismatch = QuarkusConfig.getPropertiesLanguageMismatch();
+    switch (propertiesLanguageMismatch) {
+      case PropertiesLanguageMismatch.ignore:
+        return;
+      case PropertiesLanguageMismatch.prompt:
+        const YES = "Yes", NO = "No";
+        const response: string | undefined = await window.showInformationMessage(`Do you want to force the language ID of the file  to '${languageId}' to benefit with Quarkus support?`, YES, NO);
```

Comment:
@fbricon @xorye have you some suggestion how to improve this message ?

Reply:
How about: _Do you want to recognize this file as `${languageId}` and prevent it from being recognized as a different file type? This will improve the Quarkus support._

Reply:
My idea is:
```
The language ID for ${filename} must be ${languageId} to receive language support such as
completion and validation. Set the language ID to ${languageId}?
```

![image](https://user-images.githubusercontent.com/20326645/86805209-ae6ee100-c045-11ea-8e93-319c674a5457.png)

Reply:
Thanks @xorye for your suggestion. Fixed.